app/controllers/v0/facilities/ccp_controller.rb

```diff
@@ -3,24 +3,74 @@
 class V0::Facilities::CcpController < FacilitiesController
   before_action :validate_id, only: [:show]
 
+  def index
+    raise Common::Exceptions::ParameterMissing.new('type', detail: TYPE_SERVICE_ERR) if params[:type].nil?
+
+    ppms_results =  case search_params[:type]
+                    when 'cc_provider'
+                      provider_search
+                    when 'cc_pharmacy'
+                      provider_search('services' => ['3336C0003X'])
+                    when 'cc_walkin'
+                      api.pos_locator(search_params, '17')
+                    when 'cc_urgent_care'
+                      api.pos_locator(search_params, '20')
+                    end
+
+    start_ind = (page - 1) * BaseFacility.per_page
+    ppms_results = ppms_results[start_ind, BaseFacility.per_page - 1]
+
+    render  json: ppms_results,
+            each_serializer: ProviderSerializer,
+            meta: { pagination: pages(ppms_results) }
+  end
+
   def show
-    ppms = Facilities::PPMS::Client.new
-    result = ppms.provider_info(params[:id])
+    result = api.provider_info(params[:id])
     raise Common::Exceptions::RecordNotFound, params[:id] if result.nil?
 
-    services = ppms.provider_services(params[:id])
+    services = api.provider_services(params[:id])
     result.add_provider_service(services[0]) if services.present?
     render json: result, serializer: ProviderSerializer
   end
 
   def services
-    ppms = Facilities::PPMS::Client.new
-    result = ppms.specialties
+    result = api.specialties
     render json: result
   end
 
   private
 
+  def search_params
+    params.permit(:type, :address, services: [], bbox: [])
+  end
+
+  def provider_search(options = {})
+    api.provider_locator(search_params.merge(options)).map do |provider|
+      prov_info = api.provider_info(provider['ProviderIdentifier'])
+      provider.add_details(prov_info)
+      provider
+    end
+  end
+
+  def api
+    @api ||= Facilities::PPMS::Client.new
+  end
+
+  def page
+    Integer(params[:page] || 1)
+  end
+
+  def pages(ppms_results)
+    total = ppms_results.length
+    {
+      current_page: page,
+      per_page: BaseFacility.per_page,
+      total_pages: ppms_results.length / BaseFacility.per_page + 1,
```

Comment:
Is this always an integer? maybe it needs round/floor/ceiling?

Reply:
[per_page](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/base_facility.rb#L104-L106) it is for now. I have an issue to make it a parameter.